core/trino-main/src/main/java/io/trino/sql/planner/DomainTranslator.java

```diff
@@ -502,6 +507,19 @@ protected ExtractionResult visitComparisonExpression(ComparisonExpression node,
             }
             if (symbolExpression instanceof Cast) {
                 Cast castExpression = (Cast) symbolExpression;
+                // type of expression which is then cast to type of value
+                Type castSourceType = requireNonNull(expressionTypes.get(NodeRef.of(castExpression.getExpression())), "No type for Cast source expression");
+                Type castTargetType = requireNonNull(expressionTypes.get(NodeRef.of(castExpression)), "No type for Cast source expression");
```

Comment:
`source` -> `target` in the error message?

Reply:
thanks, fixed